### PR TITLE
Implemented a credential provider for Secret Server Vault.

### PR DESCRIPTION
**Pull request by delinea-rajani**
Changes:

Implemented a credential provider for Secret Server Vault.
Rebranded all Thycotic reference to Delinea

### CODE REVIEW

File: src/main/java/com/delinea/secrets/jenkins/global/cred/SecretServerCredentials.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+		}
+
+		/**
+		 * Populates the list of available Credential IDs for the dropdown in the
+		 * Jenkins UI.
+		 *
+		 * @param item - The Jenkins item context.
+		 * @return A ListBoxModel containing the available Credential IDs.
+		 */
+		public ListBoxModel doFillCredentialIdItems(@AncestorInPath final Item item) {
```
[the remark this answers is not in the document]
Comment (delinea-rajani) [reply src/main/java/com/delinea/secrets/jenkins/global/cred/SecretServerCredentials.java:124]: resolved

File: src/main/java/com/delinea/secrets/jenkins/global/cred/SecretServerCredentials.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+			if (StringUtils.isBlank(value)) {
+				return FormValidation.error("Credential ID is required.");
+			}
+			return FormValidation.ok();
+		}
+
+		/**
+		 * Validates the Secret ID input by the user.
+		 */
+		public FormValidation doCheckSecretId(@QueryParameter final String value) throws IOException, ServletException {
```
[the remark this answers is not in the document]
Comment (delinea-rajani) [reply src/main/java/com/delinea/secrets/jenkins/global/cred/SecretServerCredentials.java:142]: resolved

[the remark this answers is not in the document]
Comment (delinea-rajani) [reply src/main/java/com/delinea/secrets/jenkins/global/cred/SecretServerCredentials.java:142]: resolved

File: src/main/java/com/delinea/secrets/jenkins/global/cred/SecretServerCredentials.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (delinea-rajani) [reply src/main/java/com/delinea/secrets/jenkins/global/cred/SecretServerCredentials.java:124]: resolved

[the remark this answers is not in the document]
Comment (delinea-rajani) [reply src/main/java/com/delinea/secrets/jenkins/global/cred/SecretServerCredentials.java:124]: resolved

[the remark this answers is not in the document]
Comment (delinea-rajani) [reply src/main/java/com/delinea/secrets/jenkins/global/cred/SecretServerCredentials.java:124]: resolved

File: src/main/java/com/delinea/secrets/jenkins/global/cred/SecretServerCredentials.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (delinea-rajani) [reply src/main/java/com/delinea/secrets/jenkins/global/cred/SecretServerCredentials.java:142]: resolved

File: src/main/java/com/delinea/secrets/jenkins/global/cred/VaultClient.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+				return usernamePassword;
+			} else {
+				return null;
+			}
+		}
+	}
+
+	public static class UsernamePassword {
+		public final String username;
+		public final String password;
```
[the remark this answers is not in the document]
Comment (delinea-rajani) [reply src/main/java/com/delinea/secrets/jenkins/global/cred/VaultClient.java:83]: resolved

**Review by delineaKrehl — APPROVED** [review]

**Pull request merged**